src/test/fuzz/addrman.cpp

```diff
@@ -25,25 +26,194 @@ void initialize_addrman()
 class CAddrManDeterministic : public CAddrMan
 {
 public:
-    void MakeDeterministic(const uint256& random_seed)
+    FuzzedDataProvider& m_fuzzed_data_provider;
+
+    explicit CAddrManDeterministic(FuzzedDataProvider& fuzzed_data_provider)
+        : m_fuzzed_data_provider(fuzzed_data_provider)
+    {
+        insecure_rand = FastRandomContext{ConsumeUInt256(fuzzed_data_provider)};
+        if (fuzzed_data_provider.ConsumeBool()) {
+            m_asmap = ConsumeRandomLengthBitVector(fuzzed_data_provider);
+            if (!SanityCheckASMap(m_asmap)) {
+                m_asmap.clear();
+            }
+        }
+    }
+
+    /**
+     * Generate a random address. Always returns a valid address.
+     */
+    CNetAddr RandAddr()
+    {
+        CNetAddr addr;
+        if (m_fuzzed_data_provider.remaining_bytes() > 0) {
```

Comment:
Maybe allow the fuzz engine to flip-flop between the two branches, if possible?

```suggestion
        if (m_fuzzed_data_provider.ConsumeBool()) {
```

Reply:
Even better!

Kept the `remaining_bytes()` check for clarity (it will still work without it because `ConsumeBool()` would always return `false` once fuzz input is exhausted but that would be somewhat unclear/obscure).